src/key.js

```diff
@@ -482,24 +485,35 @@ Key.prototype.getExpirationTime = async function() {
  * - if multiple primary users exist, returns the one with the latest self signature
  * - otherwise, returns the user with the latest self signature
  * @param  {Date} date use the given date for verification instead of the current time
+ * @param  {Object} userId (optional) user ID to get instead of the primary user, if it exists
  * @returns {Promise<{user: module:key.User,
  *                    selfCertification: module:packet.Signature}>} The primary user and the self signature
  * @async
  */
-Key.prototype.getPrimaryUser = async function(date=new Date()) {
-  // sort by primary user flag and signature creation time
-  const primaryUser = this.users.map(function(user, index) {
+Key.prototype.getPrimaryUser = async function(date=new Date(), userId={}) {
+  let users = this.users.map(function(user, index) {
     const selfCertification = getLatestSignature(user.selfCertifications, date);
     return { index, user, selfCertification };
-  }).sort(function(a, b) {
+  }).filter(({ user }) => {
+    return user.userId && (
+      (userId.name == null || user.userId.name === userId.name) &&
```

Comment:
The linter doesn't like the == null part, if you are just trying to cover the undefined and null cases, you can do !userId.name instead.